actors/evm/src/interpreter/address.rs

```diff
@@ -0,0 +1,88 @@
+use crate::interpreter::{H160, H256};
+use crate::U256;
+use fvm_shared::address::Address as FilecoinAddress;
+
+/// A Filecoin ad   dress as represented in the FEVM runtime.
+/// This type is a little bit silly now, but it will eventually handle f4 address detection.
+pub struct Address(H160);
+
+impl From<U256> for Address {
+    fn from(v: U256) -> Self {
+        Self(H256(v.into()).into())
+    }
+}
+
+impl Address {
+    /// Expect an ID address and return it if that's the case.
+    /// An ID address starts with 0xff (msb), and contains the u64 in the last 8 bytes.
+    /// We assert that everything in between are 0x00, otherwise we've gotten an illegal address.
+    ///
+    /// 0    1-11       12
+    /// 0xff \[0x00...] [id address...]
+    ///
+    /// TODO worth replacing the Option with a Result, once we figure out error handling in this actor.
+    pub fn as_id_address(&self) -> Option<FilecoinAddress> {
+        let val = &self.0 .0;
+        if (val[0] != 0xff) || !val[1..12].iter().all(|&byte| byte == 0) {
```

Comment:
bikeshed: I'd consider `[0xff; 12]` as the prefix, just to simplify a bit. But I don't feel strongly about it.

Reply:
Agree. Will keep the code as-is for now, we can easily simplify later.